internal/wasm/store.go

```diff
@@ -153,6 +158,10 @@ type (
 		Type *GlobalType
 		// Val holds a 64-bit representation of the actual value.
 		Val uint64
+
+		// address is the globaladdr(https://www.w3.org/TR/wasm-core-1/#syntax-globaladdr) of this global instance.
+		// In other words, this equals the index of this global instance in store.Globals.
```

Comment:
probably better to call this `index` and use the existing type `wasm.Index` vs the confusing uint64 which is at best an implementation detail of additions.

```suggestion
		// address is the index of this global instance in Store.Globals.
		// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#syntax-globaladdr
```